### DISCUSSION TITLE
Simple webhook to trigger password reset?

### DISCUSSION THREAD

**Discussion by poVoq** [lldap/lldap#651] [General]
I am currently looking into writing a simple IRC bot plugin to allow people to create accounts and manage their passwords.

I saw this: https://github.com/lldap/lldap/blob/main/docs/scripting.md but really for the simple functionality I need (and for general compatibility with other LDAP servers) I do not want to get into complex graphql stuff with auth tokens and all that.

However the password reset feature seems in general unauthenticated and would really help with inviting new accounts, i.e create a new account via LDAP and a random password and then trigger a password reset email for people to set their own password.

I tried use the `/reset-password/step1/user@example.org ` endpoint, as that is what the form seems to trigger, but this doesn't seem to work. I guess even though this form is unauthenticated it still uses some sort of token to prevent abuse?

I think it would be nice if there was some very simple webhook to trigger a password reset, maybe with a simple pre-set secret or so that can be passed together with the url?

Or is there already a way to do that with the current password reset feature?

**Comment by nitnelave** [comment]
You can check the request sent by the frontend when you reset your password. It may be as simple as adding the content type json header, there are no measures to prevent abuse. I'm not sure whether it works based on the email, or it requires the username, though.

**Reply by poVoq** [reply]
Ah, I see now that the actual endpoint seems to be `/auth/reset/step1/user@example.com`

But when I directly try to use that I get a 500 error and curl returns: `Internal server error: Could not send email: Connection error: Connection error: received fatal alert: ProtocolVersion`

Edit, and now it mysteriously worked via curl when using

`curl https://lldap.example.com/auth/reset/step1/user@example.com -H 'Content-Type: application/json'`

**Comment by poVoq** [comment]
I tried it with curl now and setting the content type to the same "application/json" that the normal request seems to do, but that doesn't work either.

Are you sure that there is no authorization involved? Because when I navigate to the url mentioned in the post above I get a 401 error (and get redirected to the login page), which sounds to me like there is some check involved.

**Reply by nitnelave** [reply]
Just checking: did you enable password reset? Can you reset with the web frontend?

**Reply by poVoq** [reply]
Yes it sends out emails fine if I use the web-interface.

**Reply by poVoq** [reply]
Ah wait... I see the 401 error is even happening on the main login page as it tries to query /auth/refresh

But it is still odd that it redirects to the login page instead of showing that "token has been send" message like when using the form and button.
